### PR TITLE
RFC: deduplicate __repr__ method for AMRGridPatch subclasses

### PR DESCRIPTION
**Pull request by neutrinoceros**
## PR Summary
title

### CODE REVIEW

File: yt/data_objects/index_subobjects/grid_patch.py [hunk 1]
```diff
@@ -142,7 +142,8 @@ def _setup_dx(self):
         self.dds.units = self.index.grid_left_edge.units
 
     def __repr__(self):
-        return "AMRGridPatch_%04i" % (self.id)
+        cls_name = self.__class__.__name__
+        return f"{cls_name}_{self.id:04i} ({self.ActiveDimensions})"
```
Comment (matthewturk) [inline yt/data_objects/index_subobjects/grid_patch.py:146]: I think this is fine, and don't want to change it, but I want to note that `self.ActiveDimensions` is the not set in the base class constructor.

File: yt/frontends/ytdata/data_structures.py [hunk 2]
```diff
@@ -576,9 +576,6 @@ def __init__(self, gid, index, filename=None):
         self.LeftEdge = self.index.ds.domain_left_edge
         self.RightEdge = self.index.ds.domain_right_edge
 
-    def __repr__(self):
```
Comment (matthewturk) [inline yt/frontends/ytdata/data_structures.py:579]: I don't think that this has `ActiveDimensions`, right?  So removing this override may introduce an error.

File: yt/data_objects/index_subobjects/grid_patch.py (hunk 1 above)
Comment (neutrinoceros) [reply yt/data_objects/index_subobjects/grid_patch.py:146]: Ah, my mistake. I can make it conditional then

**Comment by matthewturk** [comment]
Or just set it to None

**Comment by matthewturk** [comment]
Almost certainly, yes.

File: yt/frontends/ytdata/data_structures.py (hunk 2 above)
Comment (neutrinoceros) [reply yt/frontends/ytdata/data_structures.py:579]: This line implies (maybe erroneously) that all instances of `AMRGridPatch` have an `ActiveDimensions` attribute https://github.com/yt-project/yt/blob/0fbc997f77bf18c0b45419d4cca06a294fb09516/yt/data_objects/index_subobjects/grid_patch.py#L84

**Review by matthewturk — APPROVED** [review]

**Pull request merged**